hypershift-operator/controllers/hostedcluster/hostedcluster_controller.go

```diff
@@ -1481,7 +1488,31 @@ func (r *HostedClusterReconciler) reconcileAutoscaler(ctx context.Context, hclus
 	return nil
 }
 
-func reconcileControlPlaneOperatorDeployment(deployment *appsv1.Deployment, hc *hyperv1.HostedCluster, image string, sa *corev1.ServiceAccount) error {
+func reconcileControlPlaneOperatorDeployment(deployment *appsv1.Deployment, hc *hyperv1.HostedCluster, image string, pullSecret []byte, sa *corev1.ServiceAccount) error {
+	releaseProvider := &releaseinfo.CachedProvider{
```

Comment:
I would setup the release provider in the same way we set it up for other controllers (as an instance var):
https://github.com/openshift/hypershift/blob/main/hypershift-operator/main.go#L172-L175

Reply:
@sjenning Thanks for putting this together!

Agree with @csrwng ... I would consider changing this function signature to lift out the dependency and keep it more stateless:

```go
func reconcileControlPlaneOperatorDeployment(deployment *appsv1.Deployment, hc *hyperv1.HostedCluster, image string, *releaseinfo.ReleaseImage, sa *corev1.ServiceAccount) error
```

Setting up the release provider outside the controller, injecting it in as a member of `HostedClusterReconciler` (as @csrwng suggested), and doing the `Lookup()` call outside of `reconcileControlPlaneOperatorDeployment()`. This also gives the `Lookup()` call access to the reconciliation `Context` (so `Lookup()` can't block forever and stall the reconciler).